aten/src/ATen/test/pow_test.cpp

```diff
@@ -110,39 +110,53 @@ void assert_eq(T val, T act, T exp) {
   ASSERT_EQ(act, exp);
 }
 
-template<typename Vals, typename Pows>
-void tensor_pow_scalar(const Vals vals, c10::ScalarType vals_dtype, const Pows pows, c10::ScalarType pows_dtype) {
-  using T = typename Vals::value_type;
+template <class T,
+  typename std::enable_if<std::is_floating_point<T>::value,T>::type* = nullptr>
+T typed_pow(T base, T exp) {
+  return std::pow(base, exp);
+}
+template <class T,
+  typename std::enable_if<std::is_integral<T>::value,T>::type* = nullptr>
+T typed_pow(T base, T exp) {
+  return native::powi(base, exp);
+}
 
-  const auto tensor = torch::tensor(vals, vals_dtype);
+template<typename Vals, typename Pows>
+void tensor_pow_scalar(const Vals vals, const Pows pows, const torch::ScalarType valsDtype, const torch::ScalarType dtype) {
+  const auto tensor = torch::tensor(vals, valsDtype);
 
   for (const auto pow : pows) {
+    if ( dtype == kInt && pow > std::numeric_limits<int>::max()) {
+      // value cannot be converted to type int without overflow
+      EXPECT_THROW(tensor.pow(pow), std::runtime_error);
```

Comment:
Is this code path triggered (@ngimel and I are reading in order so maybe it is below)? We are surprised that we detect this case.

Reply:
yes, if I get rid of the EXPECT_THROW then the test results in `"value cannot be converted to type int without overflow: 3037000499" thrown in the test body.`